docs/tasks/scaleworkload.rst

```diff
@@ -0,0 +1,50 @@
+.. _scaleworkloadexample:
+
+Using ScaleWorkload function with output artifact
+-------------------------------------------------
+
+Traditionally, when ``ScaleWorkload`` functions were used to scale down the
+workload, it was required to get the original replica count of the workload and
+set it as output artifact. So that it can later be used to scale up the
+workload to same number of replicas.
+
+After the new changes, the ``ScaleWorkload`` function automatically sets the
+original replica count of the workload as output artifact, which makes using
+``ScaleWorkload`` function in blueprints a lot easier.
+
+Below is an example of how this function can be used
+
+
+.. code-block:: yaml
+
+  apiVersion: cr.kanister.io/v1alpha1
+  kind: Blueprint
+  metadata:
+    name: my-blueprint
+  actions:
+    backup:
+      outputArtifacts:
+        backupOutput:
+          keyValue:
+            origReplicas: "{{ .Phases.shutdownPod.Output.originalReplicaCount }}"
+      phases:
+      # before scaling replicas 0, ScaleWorkload will get the original replica count
+      # to set that as output artifact (originalReplicaCount)
+      - func: ScaleWorkload
```

Comment:
Ideally, we should have versioning mechanism for different varients of the function instead of overriding existing functionality. But that's separate discussion

Reply:
yeah, separate discussion and  versioning would sense, if there are no ways of consuming this function in older way and someone wants to consume the function using older way. In that case version would be useful to make sure that someone is able to specify the version and can use the older behaviour of the function.
But, in this situation people can chose to use `ScaleWorkload` in any way without the need of versioning.